core/execution/common/market_activity_tracker_checkpoint.go

```diff
@@ -105,5 +106,17 @@ func (mat *MarketActivityTracker) Load(_ context.Context, data []byte) error {
 			mat.takerFeesPaidInEpoch = append(mat.takerFeesPaidInEpoch, epochMap)
 		}
 	}
+	if b.GameEligibilityTracker != nil {
+		for _, get := range b.GameEligibilityTracker {
+			mat.eligibilityInEpoch[get.GameId] = make([]map[string]struct{}, 0, len(get.EpochEligibility))
+			for i, epoch := range get.EpochEligibility {
+				mat.eligibilityInEpoch[get.GameId][i] = make(map[string]struct{}, len(epoch.EligibleParties))
```

Comment:
I think this will panic because you've only set the capacity of `mat.eligibilityInEpoch[get.GameId]` to be `len(get.EpochEligibility)` and not its length. So when you access `[i]` it'll say its out of bounds:
https://go.dev/play/p/hxXvqf8B1bT